src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java

```diff
@@ -1570,4 +1575,123 @@ public void invalidPathNameWithValidOperator() {
+          """.formatted(UUID);
+      final ComparisonExpression expectedResult =
+          new ComparisonExpression(
+              "_id",
+              List.of(
+                  new ValueComparisonOperation(
+                      ValueComparisonOperator.EQ,
+                      new JsonLiteral(
+                          DocumentId.fromExtensionType(
+                              JsonExtensionType.UUID, objectMapper.getNodeFactory().textNode(UUID)),
+                          JsonType.DOCUMENT_ID))),
+              null);
+      FilterClause filterClause = objectMapper.readValue(json, FilterClause.class);
+      assertThat(filterClause.logicalExpression().logicalExpressions).hasSize(0);
+      assertThat(filterClause.logicalExpression().comparisonExpressions).hasSize(1);
+      assertThat(filterClause.logicalExpression().comparisonExpressions.get(0).getPath())
+          .isEqualTo(expectedResult.getPath());
+      assertThat(
+              filterClause.logicalExpression().comparisonExpressions.get(0).getFilterOperations())
+          .isEqualTo(expectedResult.getFilterOperations());
+    }
+
+    @Test
+    public void mustHandleUUIDAsRegularField() throws Exception {
+      final String UUID = "16725312-0000-0000-0000-000000000000";
+      String json = """
+            {"value": {"$uuid": "%s"}}
+          """.formatted(UUID);
+      final ComparisonExpression expectedResult =
+          new ComparisonExpression(
+              "value",
+              List.of(
+                  new ValueComparisonOperation(
+                      ValueComparisonOperator.EQ, new JsonLiteral(UUID, JsonType.STRING))),
+              null);
+      FilterClause filterClause = objectMapper.readValue(json, FilterClause.class);
+      assertThat(filterClause.logicalExpression().logicalExpressions).hasSize(0);
+      assertThat(filterClause.logicalExpression().comparisonExpressions).hasSize(1);
+      assertThat(
+              filterClause.logicalExpression().comparisonExpressions.get(0).getFilterOperations())
+          .isEqualTo(expectedResult.getFilterOperations());
+      assertThat(filterClause.logicalExpression().comparisonExpressions.get(0).getPath())
+          .isEqualTo(expectedResult.getPath());
+    }
+
+    @Test
+    public void mustFailOnBadUUIDAsId() throws Exception {
+      String json = """
+         {"_id": {"$uuid": "abc"}}
+        """;
+
+      Throwable throwable = catchThrowable(() -> objectMapper.readValue(json, FilterClause.class));
+      assertThat(throwable)
+          .isInstanceOf(JsonApiException.class)
+          .satisfies(
+              t -> {
+                assertThat(t.getMessage())
+                    .contains(
+                        "Bad JSON Extension value: '$uuid' value has to be 36-character UUID String, instead got (\"abc\")");
+              });
+    }
+
+    @Test
+    public void mustFailOnBadObjectIdAsId() throws Exception {
+      String json = """
+         {"_id": {"$objectId": "xyz"}}
```

Comment:
Minor detail: MongoDB extended JSON uses `$oid`, not `$objectId`. [Here's the internal JS implementation](https://github.com/mongodb/js-bson/blob/2ac17ec1e3c53b280efa298d137d96b2176bf046/src/extended_json.ts#L71) and [Extended JSON docs](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/#mongodb-bsontype-ObjectId).

`$oid` over `$objectId` isn't my favorite design decision, I can see the benefit of using `$objectId`. But I figured I'd surface this inconsistency so we can think a little more about whether we prefer `$oid` for consistency or `$objectId` for readability. Stargate-mongoose can do either way.

Reply:
Hmmh. I did miss this comment, for better or worse. So at least release 1.0.3 exposes `$objectId`; not sure we want to re-visit this.

/cc @amorton